### PR TITLE
add initial integration tests

### PR DESCRIPTION
**Pull request by shippo-lueders**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import os
+
+import pytest
+
+import shippo
+
+
+@pytest.fixture(scope='session')
+def api():
+    token = os.environ['SHIPPO_TOKEN']
```
Comment (shippo-beth) [inline tests/conftest.py:10]: _nit_: constant for ease of updating in tests and discoverability

File: tests/test_instalabel.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import shippo
+from shippo.models.components import Carriers, DistanceUnitTemplate, WeightUnit, \
+    AddressCreateRequest, ShipmentCreateRequest, InstantTransactionRequestBody, \
+    ParcelCreateRequest
+from tests.helpers_custom import get_carrier_account
+
+
+class TestInstalabel:
+
+    def test_instalabel(self, api: shippo.Shippo):
```
Comment (shippo-beth) [inline tests/test_instalabel.py:10]: Would it make sense to link to the documentation where this use case is described or to present a bit more details in the test name about what is being tested?

File: tests/test_instalabel.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        carrier_account = get_carrier_account(api, Carriers.USPS)
+
+        create_transaction_response = api.transactions.create_transaction(
+            request_body=InstantTransactionRequestBody(
+                carrier_account=carrier_account.object_id,
+                servicelevel_token="usps_ground_advantage",
+                shipment=ShipmentCreateRequest(
+                    address_from=AddressCreateRequest(
+                        name="Rachael",
```
Comment (shippo-beth) [inline tests/test_instalabel.py:19]: are we using the static values to match the documented use case?

File: tests/test_instalabel.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                            width="5",
+                            height="5",
+                            distance_unit=DistanceUnitTemplate.CM,
+                            weight="2",
+                            mass_unit=WeightUnit.LB
+                        )
+                    ]
+                )
+            ))
+        assert create_transaction_response.transaction is not None
```
Comment (shippo-beth) [inline tests/test_instalabel.py:50]: why not assert anything about the response?

File: tests/test_purchase_label.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                parcels=[parcel.object_id]
+            ))
+        assert create_shipment_response.shipment is not None
+        shipment = create_shipment_response.shipment
+        assert shipment.address_to.object_id == address_to.object_id
+        assert shipment.address_from.object_id == address_from.object_id
+        assert shipment.address_return.object_id == address_from.object_id
+
+        create_transaction_response = api.transactions.create_transaction(
+            request_body=TransactionCreateRequest(rate=shipment.rates[0].object_id)
```
Comment (shippo-beth) [inline tests/test_purchase_label.py:108]: this logic appears to be duplicated. consider a helper method for creating a transactions based on a shipment?

File: tests/test_purchase_label.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                weight="2",
+                mass_unit=WeightUnit.LB,
+                metadata="Wool Box1",
+            ))
+        parcel = create_parcel_response.parcel
+
+        create_shipment_response = api.shipments.create_shipment(
+            shipment_create_request=ShipmentCreateRequest(
+                address_from=address_from.object_id,
+                address_return=address_from.object_id,
```
Comment (shippo-beth) [inline tests/test_purchase_label.py:97]: why is `address_return` and `address_from` both pointing at `address_from`?

File: tests/test_rates_at_checkout.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    ServiceGroupAccountAndServiceLevel, ServiceGroupType, LiveRateCreateRequest, AddressCompleteCreateRequest, \
+    LineItem, WeightUnit, Parcel, DistanceUnitTemplate
+from tests.helpers_custom import get_carrier_account
+
+
+class TestRatesAtCheckout:
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, api: shippo.Shippo):
+        self._delete_all_service_groups(api)
```
Comment (shippo-beth) [inline tests/test_rates_at_checkout.py:14]: why is this required?

File: tests/test_rates_at_checkout.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    def _delete_all_service_groups(self, api: shippo.Shippo):
+        response = api.service_groups.list_service_groups()
+        for sg in response.service_group_list_response:
+            api.service_groups.delete_service_group(sg.object_id)
+
+    def test_create_service_group(self, api: shippo.Shippo):
+        carrier_account = get_carrier_account(api, Carriers.USPS)
+        ups_account_id = carrier_account.object_id
+
+        # TODO: update spec so that ServiceLevel* enum models are generated
```
Comment (shippo-beth) [inline tests/test_rates_at_checkout.py:25]: is this linked to a ticket in backlog?

**Review by shippo-beth — COMMENTED** [review]
Great job getting these setup! Do these automatically run as part of CI somehow?

File: tests/test_instalabel.py (hunk 3 above)
Comment (shippo-lueders) [reply tests/test_instalabel.py:19]: yeah

File: tests/test_instalabel.py (hunk 4 above)
Comment (shippo-lueders) [reply tests/test_instalabel.py:50]: b/c these tests are not meant to validate the endpoints, more about SDK usability and providing language-specific examples of major user flows.  this is effectively asserting that the response code is 201 and the transaction was successfully created

File: tests/test_purchase_label.py (hunk 5 above)
Comment (shippo-lueders) [reply tests/test_purchase_label.py:108]: I'm alright with a little duplication, the intent here is to provide a running example of the major use cases

File: tests/test_purchase_label.py (hunk 6 above)
Comment (shippo-lueders) [reply tests/test_purchase_label.py:97]: because generally the return address will be the same as where the package is coming from... at least I think that's the case

File: tests/test_rates_at_checkout.py (hunk 7 above)
Comment (shippo-lueders) [reply tests/test_rates_at_checkout.py:14]: because service group creation fails if you try to create two service groups with the same name and I'd rather no have hundreds or thousands of service groups with the same definition and different names since the service groups are used internally when generating rates

File: tests/test_rates_at_checkout.py (hunk 8 above)
Comment (shippo-lueders) [reply tests/test_rates_at_checkout.py:25]: it'll be the same ticket as this, just waiting for a spec PR to be merged

File: tests/test_instalabel.py (hunk 2 above)
Comment (shippo-lueders) [reply tests/test_instalabel.py:11]: good call, added doc links
